### ISSUE TITLE
Is pretrained model of Tacotron2 + LPCNet is avalilible?

### ISSUE THREAD

**Issue by mrgloom** [MlWoo/LPCNet#1]
Is pretrained model of Tacotron2 + LPCNet is avalilible?

**Comment by MlWoo** [comment]
We have the pretrained model of the two on mandarin dataset. But I think it is not illegal without the permission of my company to release them personally. It is not hard to train the model with your materials following the steps in readme.  I will follow the steps to rerun the training and synthesising procedures to make sure it is right.

**Comment by sheepHavingPurpleLeaf** [comment]
Hi @MlWoo, i have trained Tacotron2 about 21000 steps on a female mandarin dataset and connected to LPCNet. At 21000, the error is about 0.33 and decoder is already aligned with encoder. The output wav has really bad quality, ie. large portion of the sentence is silence and cannot tell even gender from voicing part. How many steps have you trained tacotron to achieve good sound?

**Comment by MlWoo** [comment]
I have pointed that the quality of vocoder is sensitive to the estimation of pitch parameters. maybe you could achieve it with 210000 steps. we have different params to train the tacotron2. and I think it has no much meaningful.info.for you. our loss is less than 0.1.

**Comment by sheepHavingPurpleLeaf** [comment]
@MlWoo I waited for another 10k steps, the loss stays above 0.3. Any advice on taco2 params? Thanks in advance.

**Comment by superhg2012** [comment]
Hi, @MlWoo Did you train T2 with 16k mandarin dataset?

**Comment by MlWoo** [comment]
@superhg2012  yes.

**Comment by superhg2012** [comment]
@sheepHavingPurpleLeaf which Tacotron repo did you use? any better results?

**Comment by superhg2012** [comment]
@MlWoo audio processing parameters is not used when training T2 with .f32 feature files. I tried different hparams, but can only achieve 0.2 loss, did you adjust the T2 network params? thanks in advance!

**Comment by MlWoo** [comment]
@superhg2012 The loss is gained with the teacher forcing mode.

**Comment by superhg2012** [comment]
@MlWoo thanks!! constant and scheduled, which mode is preferenced?

**Comment by MlWoo** [comment]
@superhg2012 constant mode.

**Comment by superhg2012** [comment]
many thanks !!

**Comment by superhg2012** [comment]
I trained T2 for 130k steps and the lowest loss value is 0.13, and the synthsized audio is still not good as expected.  some post processing needed?

[demo.zip](https://github.com/MlWoo/LPCNet/files/3284511/demo.zip)

@MlWoo I think LPCNet is ok, the cause is pitch parameters predicted from Tacotron2, could you give some suggestions?

**Comment by sheepHavingPurpleLeaf** [comment]
@superhg2012 Can you share your hparams? you are using pinyin to train or phoneme?

**Comment by MlWoo** [comment]
@estherxue Could you post your samples?

**Comment by superhg2012** [comment]
@sheepHavingPurpleLeaf  I used pinyin to train Tacotron2 and parameters is common.

**Comment by sheepHavingPurpleLeaf** [comment]
@superhg2012 I have got similar result with yours. Did you train LPCNet with the English dataset provided in Mozilla's repo or you used your mandarin dataset?

**Comment by superhg2012** [comment]
@sheepHavingPurpleLeaf  I used same mandarin dataset for LPCNet and Tacotron2. The sound quality is almost same while loss is around 0.13 ~ 0.17.

**Comment by estherxue** [comment]
Hi, here are my samples trained with Tacotron 2 + LPCNet.
[tacotron2+lpcnet.zip](https://github.com/MlWoo/LPCNet/files/3284754/tacotron2%2Blpcnet.zip)

**Comment by superhg2012** [comment]
@estherxue hi,  the examples sounds good, I have several questions.
1 .  you are using pinyin to train or phoneme?
2.   you are using same dataset to train both t2 and lpcnet?
3.   how many steps takes to train t2 part? and last loss?
4.  you are training in GTA mode?

thanks in advance!!

**Comment by MlWoo** [comment]
@superhg2012 Our team (Xue is my collegue) does not use any other trick to train the tacotron2.
1. pinyin
2. the same dataset
3. 280k if I remember correctly. and loss is about 0.1.  Maybe the lr scheduling is not same as the t2 repo because T2 repo is updated recently.
4. No GTA. if you want to use gta mode, there is a lot tricky work (like round audio to the frames)to be done.

**Comment by superhg2012** [comment]
@MlWoo get it, thanks for kind reply!!

**Comment by ajaysg-zz** [comment]
how long does it take to synthesize on GPU as well as CPU

**Comment by estherxue** [comment]
If my memory serves me correctly, synthesis on GPU is very slow. For
synthesis on CPU, the speed can reach above 3 times real time.

ajaysg <notifications@github.com> 于2019年10月6日周日 下午7:47写道：
